src/main/java/com/example/controllers/DashboardController.java

```diff
@@ -0,0 +1,206 @@
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.OutputStream;
+import java.security.Principal;
+import java.util.List;
+
+import javax.servlet.http.HttpServletResponse;
+
+import com.example.bean.Account;
+import com.example.bean.CashAccount;
+import com.example.bean.CreditAccount;
+import com.example.bean.FileUntrusted;
+import com.example.bean.FileUntrustedValid;
+import com.example.dao.AccountDao;
+import com.example.dao.CashAccountDao;
+import com.example.dao.CreditAccountDao;
+import com.example.facade.StorageFacade;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+@Controller
+@RequestMapping(value = "/dashboard")
+public class DashboardController {
+
+	@Autowired
+	AccountDao accountDao;
+
+	@Autowired
+	CashAccountDao cashaccountDao;
+
+	@Autowired
+	CreditAccountDao creditaccountDao;
+
+	@Autowired
+	StorageFacade storageFacade;
+
+	@RequestMapping
+	public String activity(final Model model, final Principal principal) {
+
+		Account account = accountDao.findUsersByUsername(principal.getName()).get(0);
+
+		List<CashAccount> cashAccounts = cashaccountDao.findCashAccountsByUsername(principal.getName());
+		List<CreditAccount> creditAccounts = creditaccountDao.findCreditAccountsByUsername(principal.getName());
+
+		model.addAttribute("account", account);
+		model.addAttribute("cashAccounts", cashAccounts);
+		model.addAttribute("creditAccounts", creditAccounts);
+		return "dashboard";
+	}
+
+	@RequestMapping(value = "/userDetail", method = RequestMethod.GET)
+	public String userDetail(final Model model, final Principal principal, @RequestParam(value = "username") final String username) {
+
+		Account account = accountDao.findUsersByUsername(username).get(0);
+		List<CreditAccount> creditAccounts = creditaccountDao.findCreditAccountsByUsername(principal.getName());
+
+		model.addAttribute("creditAccounts", creditAccounts);
+		model.addAttribute("account", account);
+		model.addAttribute("accountMalicious", account);
+
+		return "userDetail";
+	}
+
+	@RequestMapping(value = "/userDetail/avatar", method = RequestMethod.GET)
+	public void getImage(final HttpServletResponse response, @RequestParam(value = "image") final String image) {
+
+		File file = storageFacade.exists(image) ? storageFacade.load(image) : storageFacade.load("avatar.png");
+		try {
+			writeResponse(new FileInputStream(file), response.getOutputStream());
```

Comment:
#### Automated PR Comment From Polaris SAST
<!-- e4c4566b378dff25da3b61ee97548cba -->
<!-- Comment managed by Git Commenter, do not modify! -->
### Polaris SAST Issue - Directory Traversal
_High CWE-22_
A user-controllable string is used as part or all of a filesystem path, filename, or URI (uniform resource identifier).

An attacker may access, modify, or corrupt files that contain sensitive information or are critical to the application.

### How to fix
To protect against directory traversal the application should enforce strict access controls and validate user input used to create pathnames or create indirect object references for file access. The application should also configure access to the file system on a least privilege basis to reduce the potential impact of attacks.

Whenever possible, the application must restrict the use of user-supplied file paths.  When such paths are required for the application to function properly, indirect object references should be used for file access instead of accepting paths from the user. For example, consider an application that allows its users to specify a file to download from a directory. Instead of using a user-supplied filename to access the file, the application should maintain a mapping from integer keys to file names, ensuring that the supplied value is both an integer and corresponds to an actual file. In the case that a value does not have a corresponding file, the application should return a generic error message. This will often provide a safer way of referencing files that exist in the application than trying to check user input for malicious path manipulation.